src/coffee-script.coffee

```diff
@@ -63,6 +63,13 @@ exports.compile = compile = withPrettyErrors (code, options) ->
     token[1] for token in tokens when token[0] is 'IDENTIFIER'
   )
 
+  # Check for import or export; if found, force bare mode
+  unless options.bare? and options.bare is yes
+    for token in tokens
+      if token[0] in ['IMPORT', 'EXPORT']
+        options.bare = yes
```

Comment:
? I'm not arguing about not allowing both bare and modules. I'm arguing about doing it silently.